src/server/pwa/middleware/service-worker.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import fs from 'fs';
+
+let serviceWorkerScript = null;
+let serviceWorkerScope = null;
+let serviceWorkerType = null;
+let serviceWorkerEnabled = false;
+
+export function getServiceWorkerEnabled() {
```

Comment:
are all of these getters needed? tests should be testing behavior rather than implementation and testing these directly probably doesn't encourage looking at the behavior that is the purpose of the unit? (i.e. getting the right service worker content for a request)

Reply:
You made a valid point here. I've tightened up the config a bit and targeted testing the behavior. I think I was focusing on covering more than actual functionality.